src/backend/distributed/planner/multi_planner.c

```diff
@@ -667,6 +623,35 @@ CheckNodeIsDumpable(Node *node)
 
 
 /*
+ * CheckNodeCopyAndSerialization checks copy/dump/read functions
+ * for nodes and returns copy of the input.
+ *
+ * It is only active when assertions are enabled, otherwise it returns
+ * the input directly.
```

Comment:
Maybe add some note on why we're doing this rather silly thing. e.g.

"we use this to confirm that our serialization and copy logic produces the correct plan during regression tests"